.github/workflows/ci.yml

```diff
@@ -0,0 +1,11 @@
+name: CI
+
+on: push
```

Comment:
Let's maybe make that on pull_request ?

Reply:
No idea. `push` also works on main or other branches w/o PR. You can skip ci with a tag: https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs

Reply:
no I don't want to skip CI, I want CI to run on pull requests and to enforce a green CI before merging. I think changing it to `on: pull_request` is what we need

Reply:
applied suggestion by [lalten](https://github.com/lalten) below, which combines both